tests/Tests.elm

```diff
@@ -4,34 +4,115 @@ import Test exposing (..)
+                    \() ->
+                        let
+                            library =
+                                [ Card "a", Card "b" ]
+                        in
+                            Expect.equal ( Nothing, [ Card "a", Card "b" ] ) <|
+                                Library.Library.removeByName "c" library
+                ]
+            ]
+        , describe "adding a card"
+            [ test "to the top" <|
                 \() ->
-                    Expect.equal (3 + 7) 10
-            , test "String.left" <|
+                    let
+                        library =
+                            [ Card "a", Card "b", Card "c" ]
+                    in
+                        Expect.equal (Card "d" :: library) <|
+                            pushTop (Card "d") library
+            , test "to the bottom" <|
                 \() ->
-                    Expect.equal "a" (String.left 1 "abcdefg")
-            , test "This test should fail - you should remove it" <|
+                    let
+                        library =
+                            [ Card "a", Card "b", Card "c" ]
+                    in
+                        Expect.equal (library ++ [ Card "d" ]) <|
+                            pushBottom (Card "d") library
+            ]
+        ]
+
+
+hand : Test
+hand =
+    describe "Hand module"
+        [ describe "Adding a card" <|
+            [ test "a single card" <|
                 \() ->
-                    Expect.fail "Failed as expected!"
+                    let
+                        hand =
+                            [ Card "a", Card "b" ]
+                    in
+                        Expect.equal [ Card "c", Card "a", Card "b" ] <| add (Card "c") hand
             ]
-        , describe "Fuzz test examples, using randomly generated input"
-            [ fuzz (list int) "Lists always have positive length" <|
-                \aList ->
-                    List.length aList |> Expect.atLeast 0
-            , fuzz (list int) "Sorting a list does not change its length" <|
-                \aList ->
-                    List.sort aList |> List.length |> Expect.equal (List.length aList)
-            , fuzzWith { runs = 1000 } int "List.member will find an integer in a list containing it" <|
-                \i ->
-                    List.member i [ i ] |> Expect.true "If you see this, List.member returned False!"
-            , fuzz2 string string "The length of a string equals the sum of its substrings' lengths" <|
-                \s1 s2 ->
-                    s1 ++ s2 |> String.length |> Expect.equal (String.length s1 + String.length s2)
+        , describe "Removing a card" <|
+            [ test "by name" <|
+                \() ->
+                    let
+                        hand =
+                            [ Card "a", Card "b" ]
+                    in
+                        Expect.equal ( Just (Card "b"), [ Card "a" ] ) <| Hand.Hand.removeByName "b" hand
```

Comment:
Oh I guess now that this has been refactored out, there is no ambiguity. But when both Library and Hand were being imported into the same file, there WAS, and the qualification was required.